Desktop/components/JASP/Widgets/MainPage.qml

```diff
@@ -299,15 +310,14 @@ Item
 				{
 					request.reject();
 					console.log("Navigation requeste rejected:", requestedURL.hostname)
-
-					messages.showWarningQML(qsTr("Video link not accepted"), qsTr("JASP only allows the following videoservices: \n\n%1\n\nContact the JASP team to request adding another videoservice to the list.".arg(urlWhitelist.join(", "))));
 				}
 			}
 
 			onLoadingChanged: (loadRequest)=>
 			{
 				resultsJsInterface.resultsLoaded = loadRequest.status === WebEngineView.LoadSucceededStatus;
 				setTranslatedResultsString();
+				runJavaScript(`window.sendUrlWhitelist(${JSON.stringify(urlWhitelist)})`) //sent to js side
```

Comment:
Wouldnt it be much easier to change the whitelist once its loaded into JS in the browser?
Although I guess it isnt possible to store any JS in a jasp-file so I guess it doesnt matter?

Reply:
These will not be stored in jasp files, the whitelist will be sent to the js side via this signal (getter/setter in browser but not in node),I also used to send translation strings in qml like this, or am I missing some other way?

Reply:
Well, I mean tht if someone manages to hijack the JS a little bit they could then overwrite:
```
function isURLInWhitelist(hostname) {
	for (var i = 0; i < videoUrlList.length; i++) {
		var pattern = videoUrlList[i];
		if (pattern === hostname) {
			return true;
		} else if (pattern.indexOf('*') !== -1) {
			const regex = new RegExp(`^${pattern.replace(/\./g, '\\.').replace(/\*/g, '.*')}$`);
			if (regex.test(hostname)) {
				return true;
			}
		}
	}
	return false;
}
```
with 
```
function isURLInWhitelist(hostname) { return true; }
```
No?
And then load whatever

Reply:
aaaah wait, of course, there is 2 checks now, so yeah no that is fine!